### PR TITLE
StorageMiner.SubmitPoSt actor method charges late fees.

### PR DESCRIPTION
**Pull request by anorth**
The fee is a proportion of pledge collateral equal to the lateness as a fraction of the maximum allowable lateness. Proofs after the generation attack threshold are rejected (the miner actor can expect to be slashed).

The worker doesn't yet compute and attach any required fee, so late proofs will continue to be rejected. Part of the work for #2942.

### CODE REVIEW

File: actor/builtin/miner/miner_test.go [hunk 1]
```diff
@@ -514,6 +548,40 @@ func TestGetProofsMode(t *testing.T) {
...
+		assert.True(t, af(1).Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1010), bh(10000))))
+	})
+
+	t.Run("fee capped at total pledge", func(t *testing.T) {
+		assert.True(t, pledgeCollateral.Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(1100), bh(100))))
+		assert.True(t, pledgeCollateral.Equal(LatePoStFee(pledgeCollateral, bh(1000), bh(2000), bh(100))))
+	})
+}
+
+func af(h int64) types.AttoFIL {
```
Comment (ZenGround0) [inline actor/builtin/miner/miner_test.go]: I like these.  Wonder if they belong somewhere more generally accessible?

File: actor/builtin/miner/miner_test.go [hunk 2]
```diff
@@ -216,7 +215,7 @@ func TestMinerGetPower(t *testing.T) {
 
 		st, vms := core.CreateStorages(ctx, t)
 
-		minerAddr := createTestMinerWith(120, types.NewAttoFILFromFIL(240), t, st, vms, address.TestAddress,
+		minerAddr := createTestMinerWith(types.NewAttoFILFromFIL(240), t, st, vms, address.TestAddress,
```
Comment (shannonwells) [inline actor/builtin/miner/miner_test.go]: Why the choice of these numbers for the pledge (240 and 100 down below) and why 240 here (twice as much as before).  A comment saying that the number is enough to cover fees or not enough to cover fees or whatever the reason for that amount would be helpful.

File: actor/builtin/miner/miner_test.go [hunk 3]
```diff
@@ -326,40 +325,75 @@ func TestMinerSubmitPoSt(t *testing.T) {
...
+	proof := th.MakeRandomPoStProofForTest()
+
+	miner := state.MustGetActor(st, minerAddr)
+	minerBalance := miner.Balance
+	owner := state.MustGetActor(st, address.TestAddress)
+	ownerBalance := owner.Balance
+
+	firstCommitBlockHeight := uint64(3)
+	secondProvingPeriodStart := LargestSectorSizeProvingPeriodBlocks + firstCommitBlockHeight
+	lastSubmission := secondProvingPeriodStart + LargestSectorSizeProvingPeriodBlocks + LargestSectorGenerationAttackThresholdBlocks
```
Comment (shannonwells) [inline actor/builtin/miner/miner_test.go]: Did you intend to add `LargestSectorSizeProvingPeriodBlocks` twice to obtain lastSubmission?

File: actor/builtin/miner/miner.go [hunk 4]
```diff
@@ -867,3 +890,52 @@ func GenerationAttackTime(sectorSize *types.BytesAmount) *types.BlockHeight {
 func ProvingPeriodDuration(sectorSize *types.BytesAmount) uint64 {
 	return LargestSectorSizeProvingPeriodBlocks
 }
+
+// LatePostFee calculates the fee from pledge collateral that a miner must pay for submitting a PoSt
```
[the remark this answers is not in the document]
Comment (anorth) [reply actor/builtin/miner/miner.go]: This and CollateralForSector will have other callers, e.g. the miner worker computing the fee to attach. An alternative is to move these methods to a different place, callable from both. I opted for this so we can try to keep the actor dependencies tightly constrained (they're currently much too loose).

File: actor/builtin/miner/miner_test.go (hunk 1 above)
Comment (anorth) [reply actor/builtin/miner/miner_test.go]: I don't think so. If widely available, would need to have better names and descriptions which would defeat the purpose. It's totally acceptable for this to be duplicated a few times.

File: actor/builtin/miner/miner_test.go (hunk 2 above)
Comment (anorth) [reply actor/builtin/miner/miner_test.go]: I didn't change anything here except removing an unused first parameter "pledge".

File: actor/builtin/miner/miner_test.go (hunk 3 above)
Comment (anorth) [reply actor/builtin/miner/miner_test.go]: No, I intended to pay the maximum fee before the submission would be rejected entirely.

**Pull request merged**